flytekit/tools/translator.py

```diff
@@ -820,21 +819,73 @@ def get_serializable(
     else:
         raise ValueError(f"Non serializable type found {type(entity)} Entity {entity}")
 
-    if isinstance(entity, TaskSpec) or isinstance(entity, WorkflowSpec):
-        # 1. Check if the size of long description exceeds 16KB
-        # 2. Extract the repo URL from the git config, and assign it to the link of the source code of the description entity
-        if entity.docs and entity.docs.long_description:
-            if entity.docs.long_description.value:
-                if sys.getsizeof(entity.docs.long_description.value) > 16 * 1024 * 1024:
-                    raise ValueError(
-                        "Long Description of the flyte entity exceeds the 16KB size limit. Please specify the uri in the long description instead."
-                    )
-            entity.docs.source_code = SourceCode(link=settings.git_repo)
-    # This needs to be at the bottom not the top - i.e. dependent tasks get added before the workflow containing it
+    if (
+        isinstance(entity, (PythonTask, WorkflowBase))
+        and not isinstance(entity, ReferenceEntity)
+        and entity.module_file
+    ):
+        # Extract the repo URL from the git config, and assign it to the link of the source code of the description entity
+        entity.docs.source_code = SourceCode(link=_get_git_link(str(entity.module_file), settings))
+    # This needs to be at the bottom, not the top - i.e., dependent tasks get added before the workflow containing it
     entity_mapping[entity] = cp_entity
     return cp_entity
 
 
+def _get_git_link(module: str, settings: SerializationSettings) -> Optional[str]:
+    """
+    Get the git link from the task/workflow.
+    This is used to set the source code link in the description of the entity.
+
+    :param module: the full name of the module
+    :param settings: the serialization settings
+    :return: the git source code link
+    """
+    if not settings.git_repo:
+        return None
+
+    from flytekit.remote.remote import _get_git_root
+
+    if _is_file_pushed(settings.source_root, module):
+        return settings.git_repo + module.removeprefix(_get_git_root(settings.source_root))
```

Comment:
`removeprefix` is Python >= 3.9. (Our minimum support is 3.8)

Reply:
we just dropped support for python 3.8, so this is no longer an issue.